### PR TITLE
Report a better error message when proc_open is disallowed via disabl…

### PR DESCRIPTION
**Pull request by staabm**
…e_functions php-ini directive.

Refs #26

### CODE REVIEW

File: lib/Internal/Posix/Runner.php [hunk 1]
```diff
@@ -84,6 +84,8 @@ public function start(string $command, string $cwd = null, array $env = [], arra
             $message = "Could not start process";
             if ($error = \error_get_last()) {
                 $message .= \sprintf(" Errno: %d; %s", $error["type"], $error["message"]);
+            } elseif (!function_exists('proc_open')) {
+                $message .= "because proc_open() does not exist. Make sure its not disabled via 'disable_functions' in your php.ini";
```
Comment (kelunik) [inline lib/Internal/Posix/Runner.php]: An error is thrown if the function doesn't exist: https://3v4l.org/Z7jp5

Comment (staabm) [reply lib/Internal/Posix/Runner.php]: proc_open virtually exists but is not allowed to called from userland.

**Comment by staabm** [comment]
Reworded the error message. Maybe its more clear now

**Comment by kelunik** [comment]
```
➜ php -d disable_functions=proc_open -r 'proc_open();' 
PHP Warning:  proc_open() has been disabled for security reasons in Command line code on line 1

~ via 🐘 v7.2.5 
➜ php -d disabled_functions=proc_open -r 'var_dump(function_exists("proc_open"));'
bool(true)
```

Did the behavior change between PHP versions or why do we need that check at all? The warning is already reported to the user as can be noticed in #26.

**Comment by staabm** [comment]
As you can see in the stacktrace in https://github.com/amphp/process/issues/26#issue-318401472 you only see that the process wasnt able to start, but you dont see why.

**Comment by bwoebi** [comment]
Why should error_get_last() not give you the proper error message? That's confusing me...
```
php -d disable_functions=proc_open -r '@proc_open(); var_dump(error_get_last());'
array(4) {
  ["type"]=>
  int(2)
  ["message"]=>
  string(50) "proc_open() has been disabled for security reasons"
  ["file"]=>
  string(17) "Command line code"
  ["line"]=>
  int(1)
}
```

**Comment by staabm** [comment]
Maybe its related to some other extensions beeing loaded (blackfire, xdebug or similar).

I will be back to the VM which reproduced this problem at least in 2 weeks

**Comment by kelunik** [comment]
@staabm: Any change to get it reproduced again? Otherwise I'll close this one.

**Pull request closed without merging**